src/uuid.cr

```diff
@@ -54,6 +54,10 @@ struct UUID
     new(bytes, variant, version)
   end
 
+  def self.new(uuid : UUID, variant = nil, version = nil)
```

Comment:
Why would you want to pass existing UUID with different _variant_ or _version_, what's the use case? Shouldn't this be a no-op?

Reply:
No, this is a copy constructor. It creates a copy of another UUID but allows you to overwrite the variant or version, it replaces `variant=` and `version=`.

Reply:
I mean I have no problem having this in general.  I would not use it.
But I would not have deleted the code you did for this.

Reply:
If it's a copy constructor shouldn't the `variant` and `version` be taken from the given _uuid_?